feat: adding pre-commit.ci config (#506)
ansys/pyansys-dev-guide: doc/source/coding-style/formatting-tools.rst

``tox -e <env-name0>,<env-name1>,...``

To run all available environments, simply type ``tox``.

The ``pre-commit.ci`` tool
--------------------------

The goal of the `pre-commit.ci <https://pre-commit.ci/>`_ tool is to run the same hooks as the
``pre-commit`` tool, but in a CI environment. This tool is useful for
checking the code style of your project in a CI environment.

Although the PyAnsys ecosystem also has its own ``code-style`` action (see
`Code style action <https://actions.docs.ansys.com/version/stable/style-actions/index.html#code-style-action>`_),
the `pre-commit.ci`_ tool provides some additional features:

- It is free for public projects.
- It is compatible with any CI provider.
- It ensures that the hooks versions are up to date.
- Any changes performed by the hooks are committed back to the repository.

> - Reduces the time spent on CI by means of a caching capability on the hooks used.

Reviewer: ```suggestion
- Reduces the time spent running CI by caching the hooks used.
```
